### PR TITLE
Add overload to type() constructor for 2.7

### PR DESCRIPTION
**Pull request by kylec1**
Makes some code found in the wild (enum generation) typecheck, and matches the docs.

### CODE REVIEW

File: stubs/2.7/builtins.pyi [hunk 1]
```diff
@@ -37,7 +37,10 @@ class type:
     __module__ = ''
     __dict__ = ...  # type: Dict[unicode, Any]
 
+    @overload
     def __init__(self, o: object) -> None: ...
+    @overload
+    def __init__(self, name: str, bases: tuple, dict: Dict[str, Any]) -> None: ...
```
Comment (o11c) [inline stubs/2.7/builtins.pyi]: Should be `Sequence[type]` instead of `tuple`.

Comment (JukkaL) [inline stubs/2.7/builtins.pyi]: Actually, at least in Python 2 it has to be a tuple:
```
>>> type('foo', [str], {'x': 1})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: type() argument 2 must be tuple, not list
```

Comment (o11c) [inline stubs/2.7/builtins.pyi]: @JukkaL Except that right now, you have documented in various places that `Sequence[T]` is the way to do a homogenous tuple.

What we *really* need is `Intersection[tuple, Sequence[T]]` or something.

Comment (o11c) [inline stubs/2.7/builtins.pyi]: But until we can have both, `Sequence[type]` will allow a stronger subset of the desired type-checking.

**Comment by JukkaL** [comment]
I believe that both approaches are used in various parts of the codebase, and I think that checking here for tuple is reasonable as arguably it can be a common error to try to pass in a list. So I'm changing my mind here a little with regard to using `Sequence` for variable-length tuples.

We need to implement `Tuple[t, ...]` as specified in PEP 484 to work around the issue for good.

**Comment by o11c** [comment]
Consider code like:

    isinstance(object(), (object, object()))

If a list literal were used, this would be a runtime error immediately regardless of branch. But if the `tuple` class is used, it is only a runtime error if all the previous elements don't match.

**Pull request merged**